From 5b6226c6fa5add40cb9af67076bd61f14fddfc62 Mon Sep 17 00:00:00 2001
From: Michael Roth <michael.roth@amd.com>
Date: Tue, 26 Apr 2022 19:44:39 +0000
Subject: [PATCH 50/89] *debug: use trace_kvm_sev_es_unmap_ghcb

Signed-off-by: Michael Roth <michael.roth@amd.com>
---
 arch/x86/kvm/svm/sev.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/x86/kvm/svm/sev.c b/arch/x86/kvm/svm/sev.c
index 6c4c205091a2..38920c215fce 100644
--- a/arch/x86/kvm/svm/sev.c
+++ b/arch/x86/kvm/svm/sev.c
@@ -3160,6 +3160,24 @@ static int sev_es_validate_vmgexit(struct vcpu_svm *svm, u64 *exit_code)
 
 void sev_es_unmap_ghcb(struct vcpu_svm *svm)
 {
+	if (svm->sev_es.ghcb_sa_alloc_len >= 2)
+		trace_kvm_sev_es_unmap_ghcb(svm->sev_es.ghcb_sa,
+					    svm->sev_es.ghcb_sa_gpa,
+					    svm->sev_es.ghcb_sa_len,
+					    svm->sev_es.ghcb_sa_alloc_len,
+					    svm->sev_es.ghcb_sa_sync,
+					    svm->sev_es.ghcb_in_use,
+					    ((u8 *)svm->sev_es.ghcb_sa)[0],
+					    ((u8 *)svm->sev_es.ghcb_sa)[1]);
+	else
+		trace_kvm_sev_es_unmap_ghcb(svm->sev_es.ghcb_sa,
+					    svm->sev_es.ghcb_sa_gpa,
+					    svm->sev_es.ghcb_sa_len,
+					    svm->sev_es.ghcb_sa_alloc_len,
+					    svm->sev_es.ghcb_sa_sync,
+					    svm->sev_es.ghcb_in_use,
+					    0, 0);
+
 	/* Clear any indication that the vCPU is in a type of AP Reset Hold */
 	svm->sev_es.ap_reset_hold_type = AP_RESET_HOLD_NONE;
 
-- 
2.36.1

